### ISSUE TITLE
About mask_L1_penalty and linear layer

### ISSUE THREAD

**Issue by HomeTong** [CharlesPikachu/VideoGan#1]
Hey!There are four problems,If you can answer my question, thank you very much.

1. Why use mask_w instead of `mask` when calculating the values of `mask_L1_penalty`
2. Why use a calculation method like `tf.reduce_mean(tf.reduce_sum(tf.abs(mask_w)))` to calculate `mask_L1_penalty`
3. I noticed that you used the fully connected layer instead of the original deconvolution layer at the beginning of the forward and backward networks. Why is this set?
4. You used a in the process of generating the network to merge the forward and backward networks, `video = tf.add(foreground_masked, background_masked)` so it seems that the mask network is not used. Why do you write this?

**Comment by CharlesPikachu** [comment]
sry, `video = tf.add(foreground_masked, background_masked) `is correct.

**Comment by HomeTong** [comment]
> sry, `video = tf.add(foreground_masked, background_masked) `is correct.

Can you help me with other questions? Thank you!!!

I don't know why I didn't hit your effect after two epoch.

**Comment by CharlesPikachu** [comment]
1,2：mask_L1_penalty -> L1 regularization
Refer: https://en.wikipedia.org/wiki/Regularization_(mathematics)
3: just a trick

**Comment by HomeTong** [comment]
Thx u!! Sorry, my question is a bit more.
Can I understand that the FC layer and the convolution(or deconv) layer can achieve the same effect? 
May I ask if the design of this FC layer design just as effective in Conditional Video Generations?

**Comment by HomeTong** [comment]
I mean, does this use FC instead of CONV or DECONV to cause loss of spatial information?

**Comment by HomeTong** [comment]
Because after training 2 epoch, my effect is not the same as yours. Do you have any other parameter settings like random seed?
